### PR TITLE
libusermode, rpcmon: fix various errors

### PR DESCRIPTION
**Pull request by skvl**

### CODE REVIEW

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by skvl** [comment]
@tklengyel, is everything OK with this PR?

File: src/libusermode/userhook.cpp [hunk 1]
```diff
@@ -286,7 +286,7 @@ static dll_t* create_dll_meta(drakvuf_t drakvuf, drakvuf_trap_info* info, userho
 
 static bool make_trap(vmi_instance_t vmi, drakvuf_t drakvuf, drakvuf_trap_info* info, hook_target_entry_t* target, addr_t exec_func)
 {
-    target->pid = info->proc_data.pid;
+    target->pid = info->attached_proc_data.pid;
```
Comment (tklengyel) [inline src/libusermode/userhook.cpp:289]: I think this needs to be OS agnostic. So at least check whether the target is Windows and only then use attached_proc_data.

**Comment by tklengyel** [comment]
Largely looks fine. It's unclear to me whether `libusermode` is explicitly Windows-only right now or not. I think it shouldn't be so you should not add code that clearly will only work just with Windows.

File: src/libusermode/userhook.cpp (hunk 1 above)
Comment (skvl) [reply src/libusermode/userhook.cpp:289]: My fault! I will fix this. Thank You.

Comment (skvl) [reply src/libusermode/userhook.cpp]: Done

**Comment by skvl** [comment]
@tklengyel what's the error?

**Comment by skvl** [comment]
@tklengyel I have found other places in `libusermode` related to process information. So I've added checks there too.

**Comment by skvl** [comment]
The PR relates to #1177

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Pull request merged**